utils/settings.ts

```diff
@@ -0,0 +1,104 @@
+import { groq } from "next-sanity";
+import { sanityClient } from "./sanity";
+
+type Meta = {
+  _id: string;
+  _rev: string;
+  _type: string;
+  _createdAt: string;
+  _updatedAt: string;
+};
+
+export enum BlacklistType {
+  FP = "fp",
+  MFP = "mfp",
+}
+
+type APISettings = Meta &
+  Settings & {
+    blacklist: Array<{
+      _key: string;
+      id: number;
+      [BlacklistType.FP]: boolean;
+      [BlacklistType.MFP]: boolean;
+    }>;
+  };
+
+export type Blacklist = number[];
+
+type Blacklists = {
+  [BlacklistType.FP]: Blacklist;
+  [BlacklistType.MFP]: Blacklist;
+};
+
+export type Settings = {
+  fromDate: string;
+  toDate: string;
+  blacklists: Blacklists;
+  isTBD: boolean;
+};
```

Comment:
I see that it is not done in the rest of the project, so I'll just leave it here as a note. I would recomment looking into `sanity-typegen`, it's a neat tool that generates typescript types automatically based on the defined schemas - så you don't have to define these types and make sure to keep them updated with the schema. I think it's gotten pretty mature so it should be possible to use it in this project. I've applied it in https://github.com/abakus-ntnu/fondet, so if you want to look into that some time it's a possible source for inspiration

Reply:
That sounds really nice, would have saved some time on this PR. Would love to add that at some point!